static/app/components/duration/duration.tsx

```diff
@@ -0,0 +1,45 @@
+import type {HTMLAttributes} from 'react';
+import styled from '@emotion/styled';
+
+import formatDuration, {type Format} from 'sentry/utils/duration/formatDuration';
+import type {Duration as TDuration, Unit} from 'sentry/utils/duration/types';
+
+interface Props extends HTMLAttributes<HTMLTimeElement> {
+  /**
+   * The Duration that you want to render
+   */
+  duration: TDuration;
+
+  /**
+   * How granular to render the value. You can pass in something that has `ms`
+   * precision but only show the total number of seconds (ala Math.floor)
+   */
+  precision: Unit;
+
+  /**
+   * The style/format to render into.
+   *
+   * Default is `hh:mm:ss.sss` if the precision is `ms`
+   */
+  format?: Format;
+}
+
+const Duration = styled(({duration, format, precision, ...props}: Props) => {
+  // Style and precision should match, otherwise style will pad out missing or
+  // truncated values which we don't want in this component.
+  const style = format ?? precision === 'ms' ? 'hh:mm:ss.sss' : 'hh:mm:ss';
+
+  return (
+    <time
+      dateTime={formatDuration({duration, precision: 'ms', style: 'ISO8601'})}
+      title={formatDuration({duration, precision: 'ms', style: 'hh:mm:ss.sss'})}
+      {...props}
+    >
+      {formatDuration({duration, precision, style})}
+    </time>
+  );
+})`
+  font-variant-numeric: tabular-nums;
+`;
+
+export default Duration;
```

Comment:
shall we make this a named export?

Reply:
The develop docs don't say we're doing that, and i think it's a bad pattern because it encourages people to put multiple things inside a file which goes against this part of the develop docs: "Name the file to match the component or function that is being exported."

But we shouldn't solve that problem in this PR

Reply:
That's interesting. Although it's not in the docs, I recall us discussing a preference for named exports in a past TSC meeting. Maybe we should bring that up again

Reply:
I know how evan feels about it, which is easy to conflate with having this resvoled. But it's been in the tsc todo task list forever, it's on [github here](https://github.com/getsentry/frontend-tsc/issues/33) because i ported the topic from notion.